test/e2e_node/criproxy/runtime.go

```diff
@@ -0,0 +1,273 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package criproxy
+
+import (
+	"context"
+	"time"
+
+	internalapi "k8s.io/cri-api/pkg/apis"
+	runtimeapi "k8s.io/cri-api/pkg/apis/runtime/v1"
+)
+
+type runtimeServiceProxy struct {
+	errorInjector func(string) error
+	internalapi.RuntimeService
+}
+
+func NewRuntimeServiceProxy(client internalapi.RuntimeService) internalapi.RuntimeService {
+	return &runtimeServiceProxy{
+		RuntimeService: client,
+	}
+}
+
+func (r *runtimeServiceProxy) SetErrorInjectors(errorInjectors func(string) error) {
+	r.errorInjector = errorInjectors
+}
```

Comment:
```suggestion
func NewRuntimeServiceProxy(client internalapi.RuntimeService, errorInjector func(string) error) internalapi.RuntimeService {
	return &runtimeServiceProxy{
		RuntimeService: client,
		errorInjector: errorInjector,
	}
}
```

https://github.com/pacoxu/kubernetes/blob/ad4fab8e732da25257689ae22358cf0e1e66f345/test/e2e_node/testdeviceplugin/device-plugin.go#L48-L56

`NewRuntimeServiceProxy` returns `internalapi.RuntimeService` which has no `SetErrorInjectors` and this is not convenient when using.

Reply:
Or return the runtimeServiceProxy?

Reply:
> Or return the runtimeServiceProxy?

I think that's okay

Reply:
@pacoxu hi, since we want to control whether to use `criproxy` with `--cri-proxy-enabled`, it seems we cannot directly use `runtimeServiceProxy`. I have submitted an example of the current implementation using `criproxy`; please review it for any issues. If there are better ways to implement this, thank you for suggesting them.